test/integration/index.spec.tsx

```diff
@@ -0,0 +1,27 @@
+import mockServer from './utilities/mockServer'
+import SitePage, { getStaticPaths, getStaticProps } from '../../pages/[[...slug]]/index'
+import { render } from '@testing-library/react'
+import 'whatwg-fetch'
+
+test('render pages with dynamic path', async () => {
+  mockServer.listen({ onUnhandledRequest: 'error' })
+
+  const { paths } = await getStaticPaths()
+
+  expect(paths[0].params.slug).toBeUndefined
+  expect(paths[1].params.slug?.join('/')).toBe('demo/js')
+
+  for (const path of paths) {
+    const {
+      props: { pageContent },
+    } = await getStaticProps(path)
```

Comment:
This part is interesting. Maybe there is some ways to workaround, it seems like all integration tests for pages will go here in the single test if `[[...slug]]` is used from root.

Reply:
Maybe one test is enough for the template component.

Reply:
For me the main thing is that our testing asserts that given some defined data we get the expected output. Only really if we are doing something different with output given a different value for that data then do we need other tests / assertions. So yeah if we are mocking the data then we probably don't need to loop over it especially if the assertions / output is exactly the same.